drake/solvers/Optimization.h

```diff
@@ -0,0 +1,323 @@
+#ifndef DRAKE_OPTIMIZATION_H
+#define DRAKE_OPTIMIZATION_H
+
+#include "Core.h"
+#include <list>
+#include <memory>
+#include <initializer_list>
+
+namespace Drake {
+
+  struct DecisionVariable {
```

Comment:
I prefer to having more access control for DecisionVariable. Currently one can change the size of `DecisionVariable.value`, do we want to support that? Can we make `DecisionVariable` a class instead of a struct?